### DISCUSSION TITLE
Method SubCommand With Return Value

### DISCUSSION THREAD

**Discussion by ghost** [remkop/picocli#1803] [Q&A] [answered]
I read the documentation for subcommand as the method it seems method subcommand can't have a return value \
is it right? how can I use the return value of the method subcommand

**Comment by remkop** [comment, marked as answer]
Oh I see, the examples in the manual all return `void`.
Thanks for raising that. I will change one of them to return `int`.

If you return an `int` from your `@Command`-annotated method, that will be the exit code of that command.

**Comment by remkop** [comment]
I updated the user manual: https://github.com/remkop/picocli/commit/e3901e304add32bc9f3a4afcc460f9c07c4198bf

**Reply by ghost** [reply]
thanks for responding 🙏🏻\
is it possible to capture the return value of the method annotated with `@Command` and do something with it in the call method or somewhere else for that matter?

**Reply by remkop** [reply]
> thanks for responding 🙏🏻 is it possible to capture the return value of the method annotated with `@Command` and do something with it in the call method or somewhere else for that matter?

Hmm... the `call` method of the top-level command is not invoked if a subcommand is specified. 
If you want to run the logic of the subcommand _and_ the logic of the parent command, you can [specify the `RunAll` execution strategy](https://picocli.info/#_executing_commands_with_subcommands). However, the top-level command will not know the return value of any subcommands that were executed, so that does not do what you want.

Alternatively, you could add logic in the `main` method:

```java
CommandLine cmd = new CommandLine(new MyApp());
int exitCode = cmd.execute(args);

ParseResult parseResult = cmd.getParseResult();
List<CommandLine> all = parseResult.asCommandLineList();
CommandLine sub = all.get(all.size() - 1); // get the last subcommand
Object subcommandResult = sub.getExecutionResult();
// do stuff with subcommandResult 
```

**Reply by ghost** [reply]
this is helpful, thanks 🙏🏻💡
